test/spec/proxySpec.js

```diff
@@ -993,167 +1021,280 @@ describe("proxy", function () {
-                firstName: "Albert",
-                lastName: "Einstein",
-                phoneNumbers: [{
-                    number: "12345"
-                }, {
-                    number: "45353"
-                }]
-            };
+                observedObj.firstName = 'Malvin';            
+                expect(called).toReallyEqual(1);
 
-            var jsonPatcherProxy = new JSONPatcherProxy(obj);
-            var observedObj = jsonPatcherProxy.observe(true, function (patches) {
-                called++;
+                observedObj.firstName = 'Ronaldo';       
+                expect(called).toReallyEqual(2);
+
+                jsonPatcherProxy.pause();
+
+                observedObj.firstName = 'Messi';       
+                expect(called).toReallyEqual(2);
             });
 
-            observedObj.firstName = 'Malvin';            
-            expect(called).toReallyEqual(1);
+            it('should omit patches when unobserved then observed', function () {
+                var called = 0;
 
-            observedObj.firstName = 'Ronaldo';       
-            expect(called).toReallyEqual(2);
+                var obj = {
+                    firstName: "Albert",
+                    lastName: "Einstein",
+                    phoneNumbers: [{
+                        number: "12345"
+                    }, {
+                        number: "45353"
+                    }]
+                };
 
-            jsonPatcherProxy.switchObserverOff();
+                var jsonPatcherProxy = new JSONPatcherProxy(obj);
+                var observedObj = jsonPatcherProxy.observe(true, function (patches) {
+                    called++;
+                });
 
-            observedObj.firstName = 'Messi';       
-            expect(called).toReallyEqual(2);
-        });
+                observedObj.firstName = 'Malvin';            
+                expect(called).toReallyEqual(1);
 
-        it('should omit patches when unobserved then observed', function () {
-            var called = 0;
+                observedObj.firstName = 'Ronaldo';       
+                expect(called).toReallyEqual(2);
 
-            var obj = {
-                firstName: "Albert",
-                lastName: "Einstein",
-                phoneNumbers: [{
-                    number: "12345"
-                }, {
-                    number: "45353"
-                }]
-            };
+                jsonPatcherProxy.pause();
 
-            var jsonPatcherProxy = new JSONPatcherProxy(obj);
-            var observedObj = jsonPatcherProxy.observe(true, function (patches) {
-                called++;
+                observedObj.firstName = 'Messi';       
+                expect(called).toReallyEqual(2);
+
+                jsonPatcherProxy.resume();
+
+                observedObj.firstName = 'Carlos';       
+                expect(called).toReallyEqual(3);
             });
+            it('should unobserve then observe again (deep value)', function () {
```

Comment:
and we have `unobserve` again, It's quite confusing to me to mix `pause` and `unobserve` test together, under `describe('unobserving', function(){`